game/event/event.py

```diff
@@ -342,6 +342,8 @@ class UnitsDeliveryEvent:
     def __init__(self, control_point: ControlPoint) -> None:
         self.to_cp = control_point
         self.units: Dict[Type[UnitType], int] = {}
+        self.bought_units = {}
+        self.sold_units = {}
 
```

Comment:
These don't need to be members. Locals in `process()` or `@property` types will do.

Reply:
Made them locals, thanks!